pkg/projectfile/projectfile.go

```diff
@@ -453,6 +466,29 @@ func Parse(configFilepath string) (_ *Project, rerr error) {
 	namespace := fmt.Sprintf("%s/%s", project.parsedURL.Owner, project.parsedURL.Name)
 	StoreProjectMapping(cfg, namespace, filepath.Dir(project.Path()))
 
+	// Migrate project file if needed
+	if !migrationRunning && project.ConfigVersion != ConfigVersion && migrator != nil {
+		// Migrations may themselves utilize the projectfile package, so we have to ensure we don't start an infinite loop
+		migrationRunning = true
+		defer func() { migrationRunning = false }()
+
+		if project.ConfigVersion > ConfigVersion {
+			return nil, locale.NewInputError("err_projectfile_version_too_high")
+		}
+		updatedConfigVersion, errMigrate := migrator(project, ConfigVersion)
```

Comment:
We will lose this error if anything on line 483 fails. We may want to log it in that case if this could be valuable.

Reply:
Good catch! Perfect job for `errs.Pack()`.